src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Inflater.cs

```diff
@@ -200,6 +204,7 @@ public unsafe void SetInput(ReadOnlyMemory<byte> inputBuffer)
                 _zlibStream.NextIn = (IntPtr)_inputBufferHandle.Pointer;
                 _zlibStream.AvailIn = (uint)inputBuffer.Length;
                 _finished = false;
+                _nonEmptyInput = true;
```

Comment:
Shouldn't this be set back to false at some point? My guess is no because DeflateStream is unseekable and doens't support changing its Position, but want verify if you thought about it.

Reply:
Yes, I did think about it and even set it back to false at some point during the development of this PR. Other tests started failing though (`BrotliStreamUnitTests.WrapNullReturningTasksStream`) so I reverted to the current version. This seems necessary and correct, notably to handle 0 length data.